### ISSUE TITLE
CSV has restarted for an unknown reason v0.25.2.0019

### ISSUE THREAD

**Issue by motoridersd** [Ixam97/CarStatsViewer#203]
I recently installed this for the first time. I'm running my own test track and built the latest version following instructions found here.

The app appears to run fine when using my profile, no restarts, but if my SO drives the car, I start getting these restart messages when my profile is active again. Opening the app and letting it run properly will get rid of the error.

It seems to me like the app doesn't like it when another profile becomes active, but not sure what changes other than maybe the app is running again with a lot of missing stats from when the other driver was using the car.

What information can I provide to help debug? 

![PXL_20231101_220741160](https://github.com/Ixam97/CarStatsViewer/assets/5197858/b9827e82-e8e3-46f8-9aad-689c75e9a443)

**Comment by Ixam97** [comment]
I would not see this as a bug.

In the Center Display, every user has an own set of apps. When switching the profile the app is being suspended. When autostart is enabled the app registers a reoccuring watchdog with the operating system. This is able to detect a fatal crash, a system reboot or an update. Switching profiles is impossible to detect, so it is just a generic watchdog event after an unknown termination of the app. So it is pretty much working as intended.

**Comment by motoridersd** [comment]
Shouldn't the service be able to start automatically without notify the user about this? I understand that the profiles are separate and the app won't be running when other profiles are active, but I would expect the app to be able to restart on its own despite whatever state it might find itself in.

Or is this notification letting me know that the service already restarted? Maybe we can turn this off, because it makes a very loud sound and interrupts any music playback when this is not needed, unless the user is expected to do something.

**Comment by Ixam97** [comment]
> Shouldn't the service be able to start automatically without notify the user about this?

No. Only apps with system privileges are capable of starting background services without user interaction. The headsup-notification is a workaround to notify the user when something went wrong, preventing you from driving long distances without recording data without even noticing it. Pressing "start service" should be sufficient. Though it may take some time for this to happen and the notification might appear again. Then just ignore and wait.

You can disable this notification in the settings, but then you will have to make sure yourself whether the app is running or not. The notification should only appear when the backround service was fully stopped. You can double check this by swiping down the notifications. If there is no notification stating statsistics tracking is running, the app was fully terminated by the switching of the profile. I have not tried this myself yet since I do not have multipe profiles in my car.

If the notification fires multiple times I might change it so that it only appears once as a heads up and after that as a "silent" notification.

**Comment by motoridersd** [comment]
Gotcha. Yeah it does appear a few times as everything starts up. Is there an FAQ we can add this information to?
